src/app/components/jamjarPlayer/jamjar.js

```diff
@@ -0,0 +1,298 @@
+function JamJar(concertService, videoService, $sce) {
+  var self = this;
+
+  // stopgap until this becomes a factory with DI
+  self.concertService = concertService;
+  self.videoService = videoService;
+  self.$sce = $sce;
+
+  // concert information
+  self.concert = null;
+
+  // video that is currently playing on the screen
+  self.nowPlaying = null;
+
+  // single-source-of-truth for edges (implemented as relative edge from "first" video)
+  self.relativeEdges = {};
+
+  // hash of video_id --> video object
+  self.videosMap = {};
+
+  // list of videos that should be rendered in the view
+  self.videos = [];
+
+  self.onPlay = function() { }; // override this!
+  self.onPlayRecorded = {};
+
+  self.onFinishedJamJar = function() { }; // override this!
+
+  // default volume for videos
+  self.volume = 0.5; // 0.5
+}
+
+JamJar.prototype.initialize = function(concert_id, video_id, type, overlay) {
+  var self = this;
+
+  self.type = type;
+  self.overlay = overlay;
+
+  if (self.type == 'individual') {
+    self.loadVideo(concert_id, video_id);
+  } else if (self.type == 'jamjar') {
+    self.loadGraph(concert_id, video_id);
+  } else {
+    console.error("Invalid type given: ", type);
+  }
+};
+
+JamJar.prototype.loadVideo = function(concert_id, video_id) {
+  self.videoService.getVideoById(video_id, function(err, video) {
+    if (err) {
+      return console.error(err);
+    }
+
+    self.nowPlaying = new Video(video, {}, self.$sce);
+
+    self.videos.push(self.nowPlaying);
+  });
```

Comment:
This change for individual video player needs to be reflected on the html page too :)